aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp

```diff
@@ -0,0 +1,71 @@
+#include <c10/core/OpaqueHandle.h>
+#include <c10/core/Allocator.h>
+#include "MKLDNNCommon.h"
+
+#if AT_MKLDNN_ENABLED()
+
+#include <ideep.hpp>
+#include <caffe2/core/macros.h>
+#ifndef CAFFE2_USE_MKLDNN
+// Define IDEEP singletons here when IDEEP is not built with CAFFE2
+#include <ideep_pin_singletons.hpp>
```

Comment:
this seems backward.  If this is always built and AT_MKLDNN_ENABLED implies CAFFE2_USE_MKLDNN, then why not unconditionally put the registration in ATen and remove it from C2?  I'm trying to see if this works in https://github.com/pytorch/pytorch/pull/18335.

Reply:
@gchanan That's good. So after you merge #18335, I will remove this registration code in `MKLDNNCommon.cpp` then.

Reply:
@gchanan Removed it already, not impact the pre-checkin test anyway.